hyundai_kia_connect_api/ApiImpl.py

```diff
@@ -1,75 +1,75 @@
 import datetime as dt
 import logging
+import re
 from dataclasses import dataclass
 
 import requests
 
 from .const import *
 from .Token import Token
-from .Vehicle import Vehicle, EvChargeLimits
+from .Vehicle import Vehicle, EvChargeLimits, ClimateRequestOptions
 
 from .utils import (
     get_child_value,
-    get_hex_temp_into_index,
-    get_index_into_hex_temp,
 )
 
 _LOGGER = logging.getLogger(__name__)
 
 
-
-@dataclass
-class ClimateRequestOptions:
-    set_temp: float = None
-    duration: int = None
-    defrost: bool = None
-    climate: bool = None
-    heating: int = None
-    front_left_seat: int = None
-    front_right_seat: int = None
-    rear_left_seat: int = None
-    rear_right_seat: int = None
-
 class ApiImpl:
     data_timezone = dt.timezone.utc
     temperature_range = None
 
     def __init__(self) -> None:
         """Initialize."""
+        self.token: Token = None
 
-    def login(self, username: str, password: str) -> Token:
+    def login(self, username: str, password: str, pin: str):
         """Login into cloud endpoints and return Token"""
         pass
 
-    def get_vehicles(self, token: Token) -> list[Vehicle]:
+    def get_vehicles(self) -> list[Vehicle]:
         """Return all Vehicle instances for a given Token"""
         pass
 
-    def refresh_vehicles(self, token: Token, vehicles: list[Vehicle]) -> None:
+    def refresh_vehicles(self, vehicles: list[Vehicle]) -> None:
         """Refresh the vehicle data provided in get_vehicles. Required for Kia USA as key is session specific"""
         pass
 
     def get_last_updated_at(self, value) -> dt.datetime:
         """Convert last updated value of vehicle into into datetime"""
-        pass
-
-    def update_vehicle_with_cached_state(self, token: Token, vehicle: Vehicle) -> None:
+        if value is not None:
```

Comment:
this method is the same for all regional APIs, so I chose to factorize and put it in thep arent class (ApiImpl). The only thing I'm not sure about is this condition (if value is not None). it as present in KiaEU but not in the other classes.